generator/generator.go

```diff
@@ -545,25 +552,27 @@ func (pg *Generator) GeneratePSPWithName(
 
 	// set to default values
 	if string(psp.Spec.RunAsUser.Rule) == "" {
-		psp.Spec.RunAsUser.Rule = policyv1beta1.RunAsUserStrategyRunAsAny
+		psp.Spec.RunAsUser = pg.defaultPsp.Spec.RunAsUser
 	}
 
-	if psp.Spec.RunAsGroup != nil && string(psp.Spec.RunAsGroup.Rule) == "" {
-		psp.Spec.RunAsGroup.Rule = policyv1beta1.RunAsGroupStrategyRunAsAny
+	if psp.Spec.RunAsGroup == nil || string(psp.Spec.RunAsGroup.Rule) == "" {
+		psp.Spec.RunAsGroup = pg.defaultPsp.Spec.RunAsGroup
 	}
 
 	if string(psp.Spec.FSGroup.Rule) == "" {
-		psp.Spec.FSGroup.Rule = policyv1beta1.FSGroupStrategyRunAsAny
+		psp.Spec.FSGroup = pg.defaultPsp.Spec.FSGroup
 	}
 
 	if string(psp.Spec.SELinux.Rule) == "" {
-		psp.Spec.SELinux.Rule = policyv1beta1.SELinuxStrategyRunAsAny
+		psp.Spec.SELinux = pg.defaultPsp.Spec.SELinux
 	}
 
 	if string(psp.Spec.SupplementalGroups.Rule) == "" {
-		psp.Spec.SupplementalGroups.Rule = policyv1beta1.SupplementalGroupsStrategyRunAsAny
+		psp.Spec.SupplementalGroups = pg.defaultPsp.Spec.SupplementalGroups
```

Comment:
This prototype deals with 5 fields only. (RunAsUser, RunAsGroup, FSGroup, SELinux, SupplementalGroups.)

`volume` is not targeted.